pkg/alertmanager/operator.go

```diff
@@ -730,9 +735,13 @@ func makeSelectorLabels(name string) map[string]string {
 
 func createSSetInputHash(a monitoringv1.Alertmanager, c Config, tlsAssets *operator.ShardedSecret, s appsv1.StatefulSetSpec) (string, error) {
 	var http2 *bool
+	var serverCert string
 	if a.Spec.Web != nil && a.Spec.Web.WebConfigFileFields.HTTPConfig != nil {
 		http2 = a.Spec.Web.WebConfigFileFields.HTTPConfig.HTTP2
 	}
+	if a.Spec.ClusterTLSConfig != nil {
```

Comment:
If I'm not wrong, any change to the cluster TLS config (whether it's a struct field or the related cert/key file content on disk) should trigger a restart of the pods (e.g. Alertmanager doesn't seem to re-initiate the connections to the peers).
I'm not sure what the best option here: maybe we need `ClusterTLSConfig` to return a hash value and store it as a pod annotation?

Reply:
After more discussion, we agreed that for now we can live with the operator not restarting the pods when a parameter of the cluster TLS config changes. The idea is to move the PR forward and have a first implementation ready to use even if not perfect. We need to document the limitation though.